client/client_test.go

```diff
@@ -136,11 +137,11 @@ func TestAPI(t *testing.T) {
 
 	silOne := &types.Silence{
 		ID: "abc",
-		Matchers: []*types.Matcher{
+		Matchers: []*labels.Matcher{
 			{
-				Name:    "label1",
-				Value:   "test1",
-				IsRegex: false,
+				Name:  "label1",
+				Value: "test1",
+				Type:  labels.MatchEqual,
```

Comment:
Maybe add more tests for labels.MatchNotEqual, labels.MatchRegexp and labels.MatchNotRegexp

Reply:
This PR just moves from `types.Matcher` to `labels.Matcher`, so `labels.MatchNotEqual`and `labels.MatchNotRegexp` aren't supported yet.

In the next step, that will be done (but it also requires updates to the UI), and in that PR, you are right to ask for those tests.

Reply:
Actually, via the UI, you can probably already use the new silences, but we have no way to display them in the UI. So you could arguably already add the tests in the API tests. Mmmmh…

Reply:
I agree some tests covering silences with the new matcher types are in place. Here in `client` tests cover nothing more than the CRUD operations, though. How about some tests (mostly) regarding matching in the `silences` package, instead?

Reply:
See 4c0f5dfcc6fde0466c8d0f63630b06d38eff92c7